### PR TITLE
check dependency versions are valid on CI

### LINKED ISSUE #1532: Check deps on CI
`pip install gpflow` currently installs dependencies (setuptools, scipy) with versions that are incompatible with the tensorflow version installed.

This ticket isn't to fix the dependencies, per se, but suggests adding a `pip check -vvv` stage to CI, so that such problems are caught at PR stage.

### PR DESCRIPTION
**Pull request by joelberkeley-pio**
fixes #1532

### CODE REVIEW

**Comment by joelberkeley-pio** [comment]
> Is the dependency check executed with `READTHEDOCS` set?

pretty sure it won't. see https://docs.readthedocs.io/en/stable/faq.html#how-do-i-change-behavior-when-building-with-read-the-docs

File: setup.py [hunk 1]
```diff
@@ -12,11 +12,10 @@
 # Dependencies of GPflow
 requirements = [
     "numpy>=1.10.0",
-    "scipy>=0.18.0",
+    "scipy>=0.18.0,==1.4.1",  # pinned to ==1.4.1 to satisfy tensorflow requirements
```
Comment (insysion) [inline setup.py:15]: Johnny come lately here but we should remove the `>=0.18.0` constraint if we're going to pin it anyway. (Just to double check, is that pin strictly necessary?)

Comment (joelberkeley-pio) [reply setup.py:15]: i left it in cos they're pinned for different reasons. i don't know the reason for `>=0.18.0`

Comment (joelberkeley-pio) [reply setup.py:15]: if I we remove `>=0.18.0`, then down the line remove `==1.4.1` for some reason, we now have no constraint, while what we really want is `>=1.18.0`

**Pull request merged**